File: src/main/kotlin/com/itmo/microservices/payment/impl/entity/Payment.java
package com.itmo.microservices.payment.impl.entity;

import javax.persistence.Entity;
import javax.persistence.Id;
import java.util.Date;

@Entity
public class Payment {
    @Id
    private long paymentID;
    private long orderID;
    private long paymentCardUserID;
    private Date date;

    public Payment() {}

    public Payment(long paymentID, long orderID, long paymentCardUserID, Date date)
    {
        super();
        this.paymentID = paymentID;
        this.orderID = orderID;
        this.paymentCardUserID = paymentCardUserID;
        this.date = date;
    }

    public long getPaymentID()
    {
        return paymentID;
    }

    public void setPayment_ID(long paymentID)
    {
        this.paymentID = paymentID;
    }

    public long getOrderID()
    {
        return orderID;
    }

    public void setOrderID(long orderID)
    {
        this.orderID = orderID;
    }

    public long paymentCardUserID()
    {
        return paymentCardUserID;
    }

    public void setPaymentCardUserID(long paymentCardUserID)
    {
        this.paymentCardUserID = paymentCardUserID;
    }

    public Date getDate()
    {
        return date;
    }

    public void setDate(Date date)
    {
        this.date = date;
    }
}

File: src/main/kotlin/com/itmo/microservices/payment/impl/entity/logic/Payment.kt
import com.itmo.microservices.payment.impl.entity.api.ChangeExternalSysAcc
import com.itmo.microservices.payment.impl.entity.api.PaymentAggregate
import com.itmo.microservices.payment.impl.entity.api.PaymentReorder
import com.itmo.microservices.payment.impl.entity.api.SuccessfulPaymentEvent
import ru.quipy.domain.AggregateState
import java.lang.NumberFormatException
import java.util.*

class Payment : AggregateState<UUID, PaymentAggregate> {
    private lateinit var paymentId: UUID
    private var createdAt: Long = System.currentTimeMillis()
    private var updatedAt: Long = System.currentTimeMillis()

    private lateinit var timeAfterOrder: Date
    private lateinit var SLA: Date



    private var paymentsId = mutableMapOf<UUID, PaymentEntity>()
    override fun getId() = paymentId

    fun PaymentAggregate.orderPaymentCommand(
        userId: UUID,
        PaymentCardUser: Number
    ) : SuccessfulPaymentEvent {
        // send user data to external system account
        // check balance user

        return SuccessfulPaymentEvent(
            paymentId = paymentId
        )
    }

    fun PaymentAggregate.timeAfterOrderTimeOutCommand(
        userId: UUID,
    ) : PaymentReorder {
        // ?
        return PaymentReorder(
            paymentId = paymentId
        // id user?
        )
    }

    fun PaymentAggregate.notEnoughMoneyCommand(
        userId: UUID,
    ) : ChangeExternalSysAcc {
        // ?
        return ChangeExternalSysAcc(
            paymentId = paymentId,
            // ? id system
        )
    }

}

data class PaymentEntity (
    val id: UUID = UUID.randomUUID(),
    val orderId: Long,
    var paymentCardUserID: String,
    var price: Number,
)


